setrelease improvements

adrianschroeter opened the pull request:
Allow to set it via api and allow to opt-out default stripping via setrelease=-

adrianschroeter committed:
[api] allow to hand over setrelease via token trigger

Release command knows the setrelease parameter. Make it
also possible to use it when executing via a token.

hennevogel reviewed (changes requested):
Please add API Documentation

adrianschroeter committed:
[api] cleanup and document release api

"filter_source_repository" was just another name for "repository"
parameter.
 => Dropped it from api docu, but kept compat code to avoid
    api breakage

Clean'd up different internal method parameter namings, what could lead
to confusion about their meaning.

hennevogel commented:
> @hennevogel there's apidocs as far as I can see?

Sure thing it got added after my review. I can drop the review but it'll not make the CI go green...

hennevogel commented:
> that are timeout errors that suddenly the source server is not reachable anymore ...

There is no source server in rspec tests, just recordings of source server answers. See https://github.com/openSUSE/open-build-service/wiki/Testing-with-VCR

If you need help with this, let us know.

> these errors don't happen in our package for our production systems 

The package for our production systems does not run the specs during build, just the minitest test suite.

adrianschroeter committed:
[api] fix package check for package commands

If a token has a registered package it must not be
used of any other package.

Also don't silently skip package parameter for backend
call if package database object does not exist local.

Fixing also lacking permission check on package commands
for remote/scmsync projects.

In addition creating two test cases which can be used
to see the broken behaviour.

The new ScmSync project fixture is just a short hack
due to time limits. Will turn this into a real life
git based setup after my vacation for further tests.

adrianschroeter committed:
[api] Support lock/unlock of scmsync packages

Along with that it was need to redo the permission check code
to achieve (not only for lock and unlock) to
- operation on right object depending on command
- crash and regression fixes
- more test cases for package commands
- simplification

Co-Author: Eduardo Navarro <enavarro@suse.com>

adrianschroeter committed:
[api] fix opposite order of arguments in Package.exists_on_backend?

No reason to have only this method in opposite order

adrianschroeter committed:
[api] small simplification

Avoiding to run the same check twice.

adrianschroeter committed:
[api] remove left over code from former refactor

hopefully all covered now with previous fixes

dirkmueller commented:
@adrianschroeter there is a simple linter recommendation which could be easy to resolve for you. could you please do that?
